helpers/datasources.ts

```diff
@@ -11,6 +11,21 @@ export const EXAMPLE_FILE: TemplateDataSource = {
   },
 };
 
+export const EXAMPLE_10K_SEC_FILES: TemplateDataSource[] = [
+  {
+    type: "file",
+    config: {
+      path: "https://s2.q4cdn.com/470004039/files/doc_earnings/2023/q4/filing/_10-K-Q4-2023-As-Filed.pdf",
```

Comment:
```suggestion
      url: new URL("https://s2.q4cdn.com/470004039/files/doc_earnings/2023/q4/filing/_10-K-Q4-2023-As-Filed.pdf"),
```
you can use a union type for type safety:
```
export type FileSourceConfig =
  | {
      path: string;
    }
  | { url: URL };
  ```